lua/mini/misc.lua

```diff
@@ -409,6 +409,139 @@ MiniMisc.zoom = function(buf_id, config)
+---     ignore_filetype = { "gitcommit", "gitrebase" },
+---     center = true,
+---   }
+---   ```
+---
+---@usage >
+---   require('mini.misc').setup_restore_cursor()
+MiniMisc.setup_restore_cursor = function(opts)
+  opts = opts or {}
+  opts.ignore_buftype = opts.ignore_buftype or { "quickfix", "nofile", "help" }
+  opts.ignore_filetype = opts.ignore_filetype or { "gitcommit", "gitrebase" }
+  opts.center = opts.center or true
+
+  if not H.is_array_of(opts.ignore_buftype, H.is_string) then
+    H.error('In `setup_restore_cursor()` `opts.ignore_buftype` should be an array of strings.')
+  end
+
+  if not H.is_array_of(opts.ignore_filetype, H.is_string) then
+    H.error('In `setup_restore_cursor()` `opts.ignore_filetype` should be an array of strings.')
+  end
+
+  if not H.is_boolean(opts.center) then
+    H.error('In `setup_restore_cursor()` `opts.center` should be a boolean.')
+  end
+
+  -- local augroup = vim.api.nvim_create_augroup('MiniMiscRestoreCursor',
+  --                                             { clear = true })
+  -- local au_command = string.format("lua require('mini.misc').restore_cursor_buftype(%s)",
+  --                                  vim.inspect(opts, { newline = ' ', indent = '' }))
+  -- vim.api.nvim_create_autocmd(
+  --   'BufReadPre',
+  --   {
+  --     group = augroup,
+  --     command = au_command,
+  --     desc = 'Setup the restore cursor autocmd',
+  --   })
+
+  local au_command = string.format(
+    [[augroup MiniMiscRestoreCursor
+      au!
+      au BufReadPre * au FileType <buffer> ++once lua require('mini.misc').restore_cursor(%s)
+    augroup END]],
+    vim.inspect(opts, { newline = ' ', indent = '' }))
+  vim.api.nvim_exec(au_command, false)
+end
+
+-- MiniMisc.restore_cursor_buftype = function(opts)
+--   -- Check if the buffer should be ignored
+--   if vim.tbl_contains(opts.ignore_buftype,
+--                       vim.api.nvim_buf_get_option(0, "buftype")) then
+--     return
+--   end
+--
+--   local augroup = vim.api.nvim_create_augroup('MiniMiscRestoreCursor',
+--                                               { clear = false })
+--   local au_command = string.format("lua require('mini.misc').restore_cursor(%s)",
+--                                    vim.inspect(opts, { newline = ' ', indent = '' }))
+--   vim.api.nvim_create_autocmd(
+--     'FileType',
+--     {
+--       group = augroup,
+--       command = au_command,
+--       buffer = vim.api.nvim_get_current_buf(),
+--       once = true,
+--       desc = 'Restore the cursor position',
+--     })
+-- end
+
+--- Restore cursor when reopening a file
+MiniMisc.restore_cursor = function(opts)
+  -- Check if the buffer should be ignored
+  if vim.tbl_contains(opts.ignore_buftype,
+                      vim.api.nvim_buf_get_option(0, "buftype")) then
+    return
+  end
+
+  -- Check if the filetype should be ignored
+  if vim.tbl_contains(opts.ignore_filetype,
+                      vim.api.nvim_buf_get_option(0, "filetype")) then
```

Comment:
Can these use `vim.bo.buftype` and `vim.bo.filetype`?

Reply:
Yep, good idea.